tests/Microsoft.Identity.Test.Common/Core/Helpers/CoreAssert.cs

```diff
@@ -46,6 +47,31 @@ public static void AreEqual(DateTimeOffset expected, DateTimeOffset actual, Time
             Assert.IsTrue(DictionariesAreEqual(dict1, dict2, valueComparer));
         }
 
+        public static void IsImmutable<T>()
+        {
+            Assert.IsTrue(IsImmutable(typeof(T)));
+        }
+
+        private static bool IsImmutable(Type type)
+        {
+            if (type == typeof(string) || type.GetTypeInfo().IsPrimitive || type.GetTypeInfo().IsEnum)
```

Comment:
Well, sort of immutable - if you have primitive types, you can always change them, i.e. ImmutableType.MyInt++;